src/ucp/rndv/rndv.c

```diff
@@ -2020,6 +2027,13 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_rtr_handler,
         (sreq->send.length >=
          ep_config->am.mem_type_zcopy_thresh[sreq->send.mem_type]))
     {
+        if ((sreq->flags & UCP_REQUEST_FLAG_USER_MEMH) &&
+            !ucs_test_all_flags(sreq->send.state.dt.dt.contig.md_map,
```

Comment:
is this correct?  isn't it `sreq->send.state.dt.dt.contig.md_map` set to only `rma_bw_md_map`  rndv flow now? i think you need to call ucp_request_send_memh_eager_init here

Reply:
since param is gone we can not re-init memh by using ucp_request_send_memh_eager_init so just fallback to no pre-register if am_bw_md_map is not part of sreq->send.state.dt.dt.contig.md_map (which equals to rma_bw_md_map). Is that clear?